src/appleseed/renderer/utility/solarpositionalgorithm.h

```diff
@@ -0,0 +1,114 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright(c) 2020 Joao Marcos Costa, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// This sun positioning algorithm is based on the National Oceanic and Atmospheric
+// Administration's (NOAA) Solar Position Calculator which uses the calculations
+// presented by Jean Meeus in his book Astronomical Algorithms.
+
+// The Use of NOAA data and products are in the public domain and may be used
+// freely by the public as specify in www.nws.noaa.gov/disclaimer.php
+
+#pragma once
+
+// appleseed.renderer headers.
+#include "renderer/modeling/entity/entity.h"
+
+// appleseed.foundation headers.
+#include "foundation/memory/autoreleaseptr.h"
+#include "foundation/utility/api/specializedapiarrays.h"
+
+// Forward declarations.
+namespace foundation    { class IAbortSwitch; }
+namespace renderer      { class BaseGroup; }
+namespace renderer      { class OnFrameBeginRecorder; }
+namespace renderer      { class ParamArray; }
+namespace renderer      { class Project; }
+namespace renderer      { class ShadingContext; }
+
+namespace renderer
+{
+
+class APPLESEED_DLLSYMBOL SunPositioner
+    :public Entity
+{
+  public:
+
+    // Constructor
+    SunPositioner(
+        const char* name,
+        const ParamArray& params);
+
+    void compute_sun_position();
+    void fetch_data();
+
+    float get_zenith() const;
+    float get_azimuth() const;
+    float get_solar_noon() const;
+    float get_sunrise() const;
+    float get_sunset() const;
+
+    void SunPositioner::release()
```

Comment:
That simply won't work. This method _must_ be defined in the cpp file, otherwise whoever includes that header file gets its own implementation of `release()` which defeats the very purpose of this mechanism.